model/riscv_vmem_tlb.sail

```diff
@@ -120,3 +120,7 @@ function flush_TLB_Entry(e, asid, addr) = {
                            & not(e.global))
   }
 }
+
+function should_update_PTE_bits() -> bool = {
+  plat_enable_dirty_update() | (plat_enable_svadu() & menvcfg.HADE() == 0b1)
```

Comment:
I wonder if the logic should be changed such that `enable-dirty-update` implicitly enables svadu and sets the initial value of menvcfg.HADE to 1? In that case this function could be simplified to `plat_enable_svadu() & menvcfg.HADE() == 0b1`. I believe the --enable-dirty-update flag was only added because there was no Svadu at the time and some operating systems expect the PTE updates.

Reply:
No, because then you can't configure the model to be an implementation that does A/D updates but doesn't have Svadu, a privileged spec option which was ratified years ago.

Reply:
That is true, but looking at this PR there are no observable behaviour changes for plat_enable_svadu()/not(plat_enable_svadu()) when plat_enable_dirty_update() is also true? Does this mean we need to also guard write access to menvcfg.HADE by plat_enable_svadu()? Those older implementations should have that bit hardwired to zero (well most likely not have the register at all)?

Reply:
Yes, HADE shouldn't exist when Svadu isn't enabled, and needs to be legalised as 0

Reply:
That makes sense and I agree we should have two separate options to allow matching those implementations.